src/GraphQL.Tests/CoreToVanillaConverterTests.cs

```diff
@@ -1,8 +1,5 @@
-using System;
 using System.Threading.Tasks;
```

Comment:
I'm waiting for review about DocumentExecuter_really_big_double_valid/DocumentExecuter_really_small_double_valid

Reply:
```cs
        [Fact]
        public async Task DocumentExecuter_really_big_double_invalid()
        {
            var de = new DocumentExecuter();
            var valid = await de.ExecuteAsync(new ExecutionOptions
            {
                // create a floating-point value that is larger than double.MaxValue
                // in the expression "{double.MaxValue:0}0.0" below, the 0.0 effectively
                // multiplies double.MaxValue by 10 and the .0 forces the parser to
                // assume it is a floating point value rather than a large integer
                Query = $"{{ test(arg:{double.MaxValue:0}0.0) }}", //e.g. 1230000000.0
                Schema = Schema,
            });
            valid.ShouldNotBeNull();
            valid.Data.ShouldBeNull();
            valid.Errors.ShouldNotBeNull();
            valid.Errors[0].Message.ShouldBe($"Argument 'arg' has invalid value. Expected type 'Float', found {double.MaxValue:0}0.0.");
        }

        [Fact]
        public async Task DocumentExecuter_really_small_double_invalid()
        {
            var de = new DocumentExecuter();
            var valid = await de.ExecuteAsync(new ExecutionOptions
            {
                Query = $"{{ test(arg:{double.MinValue:0}0.0) }}", //e.g. -1230000000.0
                Schema = Schema,
            });
            valid.ShouldNotBeNull();
            valid.Data.ShouldBeNull();
            valid.Errors.ShouldNotBeNull();
            valid.Errors[0].Message.ShouldBe($"Argument 'arg' has invalid value. Expected type 'Float', found {double.MinValue:0}0.0.");
        }
```

along with this in FloatGraphType:

```cs

        /// <inheritdoc/>
        public override object? ParseLiteral(GraphQLValue value) => value switch
        {
            GraphQLIntValue x => Parse(value, x.Value),
            GraphQLFloatValue x => Parse(value, x.Value),
            GraphQLNullValue _ => null,
            _ => ThrowLiteralConversionError(value)
        };

        /// <inheritdoc/>
        public override bool CanParseLiteral(GraphQLValue value) => value switch
        {
            GraphQLIntValue x => TryParse(x.Value),
            GraphQLFloatValue x => TryParse(x.Value),
            GraphQLNullValue _ => true,
            _ => false
        };

        // IsNaN checks not really necessary because text from parser cannot represent NaN

        private double Parse(GraphQLValue value, System.ReadOnlySpan<char> chars)
        {
            var number = Double.Parse(chars);
            if (/* double.IsNaN(number) || */ double.IsInfinity(number))
                ThrowLiteralConversionError(value);
            return number;
        }

        private bool TryParse(System.ReadOnlySpan<char> chars)
            => Double.TryParse(chars, out var number) /* && !double.IsNaN(number) */ && !double.IsInfinity(number);
```

Reply:
This is a spec-compliance issue aka bug fix, and should be noted in the migration doc.

Reply:
Thanks. I added tests. One more thing - NET48 throws `OverflowException` in `BadFloatValues` test. Maybe to catch it and rethrow `ThrowLiteralConversionError` ?

Reply:
> Thanks. I added tests. One more thing - NET48 throws `OverflowException` in `BadFloatValues` test. Maybe to catch it and rethrow `ThrowLiteralConversionError` ?

I knew it would. I’d just leave it; only the error message changes per platform. But you can change it if you like.

Reply:
Or you can throw overflow exception on newer platforms which would be more descriptive of the problem. But I’d just leave it.

Reply:
🤔 I chose to catch and append line to message:
```
Unable to convert '1797693134862320000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000.0' literal from AST representation to the scalar type 'Float'; Value was either too large or too small for a Double.
```

See changes in `ScalarGraphType.ThrowLiteralConversionError`

Reply:
> perhaps the rest of the tests in this file should be moved.

Renamed in `FloatGraphTypeBadValueTests.cs` and moved into `Types` folder.